### PR TITLE
Add cluster dump to post-processing block after AT runs

### PR DESCRIPTION
**Pull request by michael-cico**
# Description

Add cluster dump to post-processing block after AT runs
# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/upgrade/Jenkinsfile [hunk 1]
```diff
@@ -424,15 +438,15 @@ pipeline {
                     script {
                         METRICS_PUSHED=metricTimerEnd("${VZ_TEST_METRIC}", '0')
                         if ( fileExists(env.TESTS_EXECUTED_FILE) ) {
-                            dumpK8sCluster('upgrade-failure-post-run-cluster-dump')
+                            dumpK8sCluster('upgrade-run-failed-post-run-cluster-dump')
```
Comment (pfmackin) [inline ci/upgrade/Jenkinsfile:441]: post-uninstall might be better

File: ci/upgrade/Jenkinsfile [hunk 2]
```diff
@@ -359,6 +359,20 @@ pipeline {
                             archiveArtifacts artifacts: '**/coverage.html,**/logs/*', allowEmptyArchive: true
                             junit testResults: '**/*test-result.xml', allowEmptyResults: true
                         }
+                        failure {
+                            script {
+                                if ( fileExists(env.TESTS_EXECUTED_FILE) ) {
```
Comment (pfmackin) [inline ci/upgrade/Jenkinsfile:364]: why does this ENV VAR trigger a dump?

File: ci/upgrade/Jenkinsfile (hunk 1 above)
Comment (mbenmclean) [reply ci/upgrade/Jenkinsfile:441]: I thought the point of the change was to dump the cluster before the uninstall to pick up the contents of pods and stuff

**Review by mbenmclean — DISMISSED** [review]

File: ci/upgrade/Jenkinsfile (hunk 1 above)
Comment (michael-cico) [reply ci/upgrade/Jenkinsfile:441]: It does.  This is an additional post-uninstall dump. It's in the above snippet.

File: ci/upgrade/Jenkinsfile (hunk 2 above)
Comment (michael-cico) [reply ci/upgrade/Jenkinsfile:364]: not sure, it was there?  I think it is for verifying that tests ran before we dump anything.

File: ci/upgrade/Jenkinsfile (hunk 1 above)
Comment (michael-cico) [reply ci/upgrade/Jenkinsfile]: Took a pass at simplifying the names and pushed

**Review by pfmackin — APPROVED** [review]

**Pull request merged**